### ISSUE TITLE
feat: Add Svelte and Vue Parsers to Oxc

### ISSUE THREAD

**Issue by baseballyama** [oxc-project/oxc#8171]
## Background
Currently `oxlint` only supports the `<script>` part of the svelte/vue file.
This is useful, but the `no_unused_vars` rule, for example, does not work correctly. Therefore, the rule itself must be turned off at this time. (I've created https://github.com/oxc-project/oxc/pull/8170)

For example, the following code will report `no_unused_vars` error but this is false positive. 
```svelte
<script>
// Svelte component
let foo = $state(0);
</script>

{foo}
```

By adding support for Svelte and Vue, we can solve these issues and provide nicer linting and formatting.

## Proposal
This proposal aims to achieve the following goals:

1. **Add Svelte Parser**: Introduce a parser that supports Svelte 5. The goal is to make the parser available as an external package at this stage (for example, use it from [inclusion-vapor](https://github.com/kazupon/inclusion-vapor)).  
2. **Migrate eslint-plugin-svelte**: Gradually migrate rules related to Svelte 5.  
3. **Support for Formatter**: Once the rework of the formatter is complete, add support for Svelte files (this will likely involve migrating `prettier-plugin-svelte`).  
4. **Vue Support**: If the approach works well for Svelte, expand it to support Vue as well.  

## Out of Scope
At this stage, migrating the Svelte compiler itself is out of scope.

## Additional Notes
The current documentation states that there are no plans to officially support `eslint-plugin-svelte` or `eslint-plugin-vue`. However, I believe this contribution can add value to oxc.

I plan to start working on this once I get a go sign from @Boshen or other core team members. (Since this is likely not part of void0's milestones, I intend to work on it gradually at my own pace.)  
For the parser, I plan to submit small, incremental PRs rather than one large PR to make the process more manageable.  
If there are any policies or guidelines regarding how to proceed, I’d appreciate it if you could share them with me.

**Comment by Boshen** [comment]
The oxc project does not have the bandwidth to maintain additional parsers as our focus is ECMAScript, but I'm happy make oxlint depend on a well maintained crate.

**Comment by Boshen** [comment]
Try and join forces and establish your own projects, for example https://github.com/oxc-project/oxc/issues/8544

**Comment by baseballyama** [comment]
@Boshen 

Nice! Let's do it. I am currently working on `eslint-plugin-svelte` v3 and will work on it as soon as that is finished.👍

**Comment by xeho91** [comment]
@baseballyama May I ask if you have started anything yet? I'm willing to lend a hand regarding parsing the Svelte-related AST nodes.

I'm reachable via [Discord](https://discord.com/users/xeho91) - I've joined oxc Discord server as well.

**Comment by baseballyama** [comment]
Hello! @xeho91  (`svelte-ast-print` is incredibly useful! Thank you so much!)

Unfortunately, due to other commitments, I haven’t been able to start yet. My initial goal is to convert part of a template into an AST and integrate the simplest oxlint rule into oxc.

I’m not sure if it’s feasible to collaborate with multiple people at such an early stage, but we could discuss the overall design before diving into coding. Also, since I’m not very familiar with Rust, I think it would be great to work with someone who is more experienced in it.

Would it be a good idea to start by having a discussion on the oxc Discord? If you already have thoughts on the architecture, I’d love to hear them.
For example, I’d like to clarify how AST types should be handled in oxlint and whether we need to implement a custom scope manager.
